Add documentation for remote debugging with pdb

ivonastojanovic opened the pull request:
Adds a brief note to the pdb help summary about attaching to a running process using the -p option, making the remote debugging feature more visible.

vstinner reviewed (approved):
LGTM

vstinner reviewed (commented):
Oh, `-m pid` is not documented at https://docs.python.org/dev/library/pdb.html. You may update this option there as well.

ivonastojanovic committed:
Mention remote debugging via -p PID in usage text

Adds a brief note to the pdb help summary about attaching to a running
process using the -p option, making the remote debugging feature
more visible.

gaogaotiantian commented on Doc/library/pdb.rst:
```diff
@@ -16,7 +16,9 @@ The module :mod:`pdb` defines an interactive source code debugger for Python
-supports post-mortem debugging and can be called under program control.
+supports post-mortem debugging and can be called under program control. The
+:mod:`pdb` module also supports remote attaching to a running Python process
+using a ``-p PID`` command-line option.
```
I don't think this is the place to put the docs. It's really weird to mention a command at the first introduction paragraph. Let's revert this piece and just talk about it later. Or, if you really want this to be the shining feature, we can add "remote debugging" after "post-mortem debugging". That's about the amount of this feature I think appropriate in the first paragraph.

ivonastojanovic replied:
We can remove it, I wasn’t convinced it should be there to begin with

gaogaotiantian commented on Doc/library/pdb.rst:
```diff
@@ -80,7 +82,10 @@ The debugger's prompt is ``(Pdb)``, which is the indicator that you are in debug
-   python -m pdb [-c command] (-m module | pyfile) [args ...]
+   python -m pdb [-c command] (-m module | -p pid | pyfile) [args ...]
+
+Remote debugging is supported by attaching to a running Python process using
```
This line is also not necessary because it was immediately mentioned later. I think a good way to think is - is `-m` mentioned in the paragraph? I think of `-p`, `-m` and `file` as the same concept level - they are debugging targets. So wherever `-m` is mentioned, `-p` should probably be mentioned.

gaogaotiantian commented on Doc/library/pdb.rst:
```diff
@@ -104,6 +109,27 @@ useful than quitting the debugger upon program's exit.
+
+   Attach to the process with the specified PID.
+
+   .. versionchanged:: 3.14
```
This should be `.. versionadded` and you don't need this much description. You don't have to introduce the background story of it. For example, `sys.settrace` is never mentioned before 3.13 (they were only added because the user can change the backend now). I think a simple `versionadded` is enough.

gaogaotiantian commented on Doc/library/pdb.rst:
```diff
@@ -104,6 +109,27 @@ useful than quitting the debugger upon program's exit.
+debug it interactively. Notice that due to how the Python interpreter works,
+attaching to a remote process that is blocked in a system call or waiting for
+I/O will only work once the next bytecode instruction is executed or when the
+process receives a signal.
```
We are a bit wordy here. I think "Attach to the process with the specified PID." already does the job perfectly. Most of the docs here just repeat that sentence in a more verbose way. The I/O part could be helpful, which we can put in a `.. note` section.

ivonastojanovic committed:
Mention remote debugging in pdb.rst

vstinner reviewed (approved):
LGTM. Thanks for updating pdb doc as well.
